cmd/driver-registrar/main.go

```diff
@@ -110,6 +224,9 @@ func main() {
 	}
 	glog.V(2).Infof("CSI driver node ID: %q", csiDriverNodeId)
 
+	if *kubeletRegistrationPath != "" {
+		k8sRegistration(csiDriverName)
```

Comment:
do we need this to "finish" before going on? right now its really hard to tell whether this is a blocking or non-blocking function. Could you make that clearer either by naming or cleaning up the code.

Reply:
Make it explicit at this level in `main()` whether this function is blocking/non-blocking and if it terminates the rest of the function or not. Does it wait for a certain condition? block forever? it is very hard to tell both actual behavior AND intention

Reply:
done